aztk/internal/cluster_data/node_data.py

```diff
@@ -147,7 +146,8 @@ def _add_plugins(self):
                     execute='{0}/{1}'.format(plugin.name, plugin.execute),
                     args=plugin.args,
                     env=plugin.env,
-                    runOn=plugin.run_on.value,
+                    target=plugin.target.value,
+                    target_role=plugin.target_role.value,
```

Comment:
This is a breaking change. All existing plugins use the runOn field which breaks them. This change should update all existing plusings and maybe deprecate runOn instead of remove it for now?

Reply:
Since plugins don't exist in the latest release (v0.6.0), this shouldn't be a breaking change. I think going forward we should limit deprecation to the cross release changes. 

Anyone using plugins today is using master, and should not expect the same level of stability as if they were upgrading from a previous minor release.